test/ex_machina/ecto_test.exs

```diff
@@ -44,18 +45,38 @@ defmodule ExMachina.EctoTest do
     }
   end
 
-  test "params_for/2 raises when passed a map" do
-    assert_raise ArgumentError, fn ->
-      TestFactory.params_for(:user_map)
-    end
-  end
-
   test "params_for/2 removes fields with nil values" do
     assert TestFactory.params_for(:user, admin: nil) == %{
-      name: "John Doe"
+      name: "John Doe",
+      articles: [],
     }
   end
 
+  test "params_for/2 recursively strips has_many values" do
+    article = TestFactory.build(:article)
+
+    user_params = TestFactory.params_for(:user, articles: [article])
+
+    assert user_params[:articles] == [%{
```

Comment:
Sorry, I think I've misunderstood the description!

The `has_many` association of `User`, the `:articles` is expected to be built, but the _recursive associations_ (eg. `:editor` of `Article`) should not - is this the purpose of this test? In such a case, would the test description `"builds one level of has_many association"` be more helpful? And, as a proof, maybe test like:

```elixir
article = TestFactory.build(:article, editor: build(:user)) # (1). I'm setting editor here explicitly...

user_params = TestFactory.params_for(:user, articles: [article])

assert user_params[:articles] == [%{
  title: article.title,
  editor: nil # (2). ... and proof it's been striped as the second level association
}]
```

What do you think? Would this reveal the intention of the functionality?